coap-core/src/main/java/com/mbed/coap/server/internal/CoapUdpMessaging.java

```diff
@@ -103,10 +108,43 @@ public final void init(int duplicationListSize,
             long delayedTransactionTimeout,
             DuplicatedCoapMessageCallback duplicatedCoapMessageCallback) {
 
+        if (duplicationListSize > 0) {
+            cache = new DefaultDuplicateDetectorCache<CoapRequestId, CoapPacket>(
+                    "Default cache",
+                    duplicationListSize,
+                    duplicateMsgDetectionTimeMillis,
+                    duplicateMsgCleanIntervalMillis,
+                    duplicateMsgWarningMessageIntervalMillis,
+                    scheduledExecutor);
+        }
+        init(cache,
+                isSelfCreatedExecutor,
+                idContext,
+                maxQueueSize,
+                defaultPriority,
+                delayedTransactionTimeout,
+                duplicatedCoapMessageCallback,
+                scheduledExecutor);
+
+
+    }
+
+    public final void init(
+            PutOnlyMap<CoapRequestId, CoapPacket> cache,
+            boolean isSelfCreatedExecutor,
+            MessageIdSupplier idContext,
+            int maxQueueSize,
+            CoapTransaction.Priority defaultPriority,
+            long delayedTransactionTimeout,
+            DuplicatedCoapMessageCallback duplicatedCoapMessageCallback,
+            ScheduledExecutorService scheduledExecutor
+    ) {
         if (coapTransporter == null || scheduledExecutor == null || idContext == null || defaultPriority == null || duplicatedCoapMessageCallback == null) {
             throw new NullPointerException();
         }
-
+        if (cache != null) {
```

Comment:
make sure that the cache is always created by the builder. This class is internal, no need to support the case when cache is null.

Reply:
Done.